code/datums/components/edible.dm

```diff
@@ -86,13 +86,19 @@ Behavior that's still missing from this component that original food items had t
 ///All the checks for the act of eating itself and
 /datum/component/edible/proc/TryToEat(mob/living/eater, mob/living/feeder)
 
+	set waitfor = FALSE

```

Comment:
@AnturK Essentialy, due to how signals work we need to set waitfor False so if the proc runs sleep() (e.g., in a do_after like we do in this component) it will return the current return value instead of not returning anything. (Which would result in you hitting urself with the food instead of eating it because the return value says you shouldnt do an attack)